### ISSUE TITLE
Total run Count is not being updated after a task run

### ISSUE THREAD

**Issue by elyssonmr** [celery/django-celery-beat#60]
Hello guys, I need your help with an issue (this could be an issue with django_celery_beat)
I'm going to use celery and django celery beat in a project with many async tasks.
In my personal tests I realized that _PeriodicTask.total_run_count_ is not being updated after every run. There is a reason for that? Maybe I'm missing some concept.
I'm sending the periodic task ID to my task in order to verify if it already executed a 100 times (for example) and I get some executions with same _total_run_count_ (not sure if this is a good pratice).

**Task Code Example:**
`
@celery_app.task
def print_run_count(*args, **kwargs):
    tks = PeriodicTask.objects.get(pk=kwargs['periodic_task'])
    print(tks.total_run_count)
`
Thanks

**Comment by elyssonmr** [comment]
**I forgot the terminal output:**

`
[2017-07-10 23:07:20,467: INFO/MainProcess] Received task: test_tasks.tasks.print_run_count[402d0447-54e2-4da7-a78f-5d0b81b63893]  
[2017-07-10 23:07:22,083: WARNING/PoolWorker-4] 99
[2017-07-10 23:07:22,084: INFO/PoolWorker-4] Task test_tasks.tasks.print_run_count[402d0447-54e2-4da7-a78f-5d0b81b63893] succeeded in 0.004580827999234316s: None
[2017-07-10 23:07:30,472: INFO/MainProcess] Received task: test_tasks.tasks.print_run_object[c0147985-1b65-4eb8-a54b-0c02848cff82]  
[2017-07-10 23:07:32,086: WARNING/PoolWorker-1] 114
`

**Comment by auvipy** [comment]
really sorry that we couldn't respond to you in time. should we close the issue or this might be still an issue?

**Comment by elyssonmr** [comment]
I think it still a issue. But I am not using Celery anymore :(

**Comment by auvipy** [comment]
I will try to verify but no other issues reported like this

**Comment by salykin** [comment]
Hi.

I have the same issue.
I want to get the actual count of task launches at any moment from within any worker. I try to get this metric from the `total_run_count` of the PeriodicTask model. 
I set a schedule to run my task every minute. I have inspected, that `total_run_count` field updates not after each task launch, but after "Writing entries..." log record of the Beat scheduler:
![изображение](https://user-images.githubusercontent.com/2499169/134712181-79fcb9f2-1fab-4adb-b6d7-0359a2f05f45.png)

In my case flushing updates occurred every 3 minutes. It is reasonable for caching and reducing a count of db-queries. 

Is there a way to obtain the real count of task launches from a worker?
